### PR TITLE
Update ButtonIcon to support style prop

### PR DESCRIPTION
**Pull request by burkeshartsis**
## DESCRIPTION
Update ButtonIcon to support additional styles passed somewhere we can render them appropriately.

### What does this PR do, or why is it needed?
We needed to be able to render a `backgroundColor` on `ButtonIcon`. This makes that possible among other things.

### How do I test this PR?
Add a style to a currently in use `ButtonIcon` somewhere in the app. Easiest place is in the media player.

### What automated tests did you write?
🍪's 

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by conrad-vanl** [comment]
1. Why can't we use `style`?
2. Do the other `Button` components support `backgroundColor`?

**Comment by burkeshartsis** [comment]
Great questions @conrad-vanl! Answers below,

1. Originally I had avoided the using `style` because I would have to pass it to a child component and not the parent to get it to work correctly. Which would could lead to unexpected results for a future developer. I don't mind this approach this was just my only concern with it.

2. Other buttons do not have a similarly named prop. In fact, to you point in above I have always advocated for styles being passed as a `style` rather than extra api. 

The PR now reflects your suggestion.

File: packages/apollos-ui-kit/src/Button/ButtonIcon/index.js [hunk 1]
```diff
@@ -52,6 +55,10 @@ class ButtonIcon extends PureComponent {
               ? this.props.iconPadding
               : this.props.size
           }
+          additionalStyles={this.props.style}
+          borderRadius={
+            this.props.iconPadding + this.props.size || this.props.size * 2
```
Comment (redreceipt) [inline packages/apollos-ui-kit/src/Button/ButtonIcon/index.js]: Why would want we want to double the size of the icon footprint implicitly if they don't pass in anything? Shouldn't we just let them decide that by letting the footprint be exactly the size they specified?

**Comment by redreceipt** [comment]
@burkeshartsis not sure if I'm testing this right but shouldn't the tab icons turn colors here?
Screen Shot 2019-03-20 at 10 57 34 AM

File: packages/apollos-ui-kit/src/Button/ButtonIcon/index.js (hunk 1 above)
Comment (burkeshartsis) [reply packages/apollos-ui-kit/src/Button/ButtonIcon/index.js]: One word, tappability. We want to provide a minimum level of usability. The available api allows for customizability in size while preserving this minimum level of ux sugar.

**Comment by burkeshartsis** [comment]
@redreceipt your screenshot is using `Icon` not `ButtonIcon`

**Comment by burkeshartsis** [comment]
@redreceipt also this just exposes the `style` prop for the `ButtonIcon` you don't pass a `backgroundColor` directly you have to pass it via a style object. Check the tests or stories for an example.

**Pull request merged**